kong/clustering/compat/init.lua

```diff
@@ -334,35 +387,9 @@ function _M.update_compatible_payload(payload, dp_version, log_suffix)
     end
   end
 
-  if dp_version_num < 3002000000 --[[ 3.2.0.0 ]] then
-    local config_plugins = config_table["plugins"]
-    if config_plugins then
-      for _, plugin in ipairs(config_plugins) do
-        if plugin["instance_name"] ~= nil then
-          ngx_log(ngx_WARN, _log_prefix, "Kong Gateway v" .. KONG_VERSION ..
-            " contains configuration 'plugin.instance_name'",
-            " which is incompatible with dataplane version " .. dp_version .. " and will",
-            " be removed.", log_suffix)
-          plugin["instance_name"] = nil
-          has_update = true
-        end
-      end
-    end
-  end
-
-  if dp_version_num < 3001000000 --[[ 3.1.0.0 ]] then
-    local config_upstream = config_table["upstreams"]
-    if config_upstream then
-      for _, t in ipairs(config_upstream) do
-        if t["use_srv_name"] ~= nil then
-          ngx_log(ngx_WARN, _log_prefix, "Kong Gateway v" .. KONG_VERSION ..
-                  " contains configuration 'upstream.use_srv_name'",
-                  " which is incompatible with dataplane version " .. dp_version .. " and will",
-                  " be removed.", log_suffix)
-          t["use_srv_name"] = nil
-          has_update = true
-        end
-      end
+  for ver, fn in pairs(compatible_checkers) do
```

Comment:
I'm not sure that `pairs()` is sufficient for this. There is an expectation that the compatibility change functions are executed in descending order (`3.2.0.0`, then `3.1.0.0`, then `3.0.0.0`, etc).

Curious, have you taken a look at the [compat code in koko](https://github.com/Kong/koko/blob/1b5cdf241279f7a49d8c637f9bc43babd7e35ae8/internal/server/kong/ws/config/compat/compatibility_table.go#L95-L1047)? It's not without its quirks, but I think this would be a good pattern for us to follow--pretty big task to implement though.

Reply:
Thanks for your opinion. I have changed the hash table to an array.

Do you think it is OK?